### DISCUSSION TITLE
BLE GATT client app

### DISCUSSION THREAD

**Discussion by linardsee** [unoplatform/uno#11262] [General]
Hello!
I would like to implement BLE GATT client application which is able to create a connection with my embedded device and communicate via GATT protocol. The target platforms are Android and iOS. What libraries should I use? Seems there is no support in the UNO documentation yet according to this: https://platform.uno/docs/articles/features/bluetoothdevice.html

Can someone give me some ideas on how to get started?

**Comment by pkar70** [comment]
There was some impulse to write Bluetooth support, but someone stops this (as such support appears to be title of someone thesis). I can implement Bluetooth in Uno, but only for Android.

**Reply by YuyhoMarduk** [reply]
Hi pkar, im trying to implement bluetooth for my android app in unoplatform and im finding it quite hard. Is for my bachelor final project, any recomendations or documentation?

Pd: Love krakow, did my erasmus there.

**Comment by jeromelaban** [comment]
We do not have a timeline to implement Bluetooth at this time, but you can follow https://github.com/unoplatform/uno/issues/7710 for updates.

**Comment by sasakrsmanovic** [comment]
We've heard of people using https://shinylib.net/ if that helps.

**Comment by mtmattei** [comment]
We just released a blog that might be helpful to you. 
[Uno Platform and Bluetooth LE: An Easy-to-Follow Guide to Building a Bluetooth Explorer](https://platform.uno/blog/building-a-bluetooth-explore-with-uno-platform/)

**Comment by pkar70** [comment]
And can we use this Nuget in Uno, and simply add converter layer between UWP API and BT nuget API?

**Comment by pkar70** [comment]
Would you like to make PR adding BLE support to Uno, or add BT coded in
'plain Android' to your app?
Which UWP methods/classes do you use / want to use?

You can always visit Krakow again :) which university you 'erasmused'?

On Thursday, June 22, 2023, YuyhoMarduk ***@***.***> wrote:
> Hi pkar, im trying to implement bluetooth for my android app in
unoplatform and im finding it quite hard. Is for my bachelor final project,
any recomendations or documentation?
>
> Pd: Love krakow, did my erasmus there.
>
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you commented.<
https://ci5.googleusercontent.com/proxy/FiiJx4jHr48roN2wa7rjomTpeVJUyXeREL7Up3sBi3c9Xem6gfKceuWaPjjvrzZXtBykD-oYyWstCplI6LFuZKmiTmMnXimRLI-ZgNLMVEccotQccig6us_e7-gkh8oYrsdnS_Bbniu26Dhk8f3mz7KPLPsi8bAlDgCxs4VXy18WCc6GF6xmzyNfR72UYDn0ZIAVyXTL2Urj0nNntiWKwrV4D_XNm8hyJytqV6NC43-BPn9G0NVqj7xuAUKM=s0-d-e1-ft#https://github.com/notifications/beacon/AFS5O4ZNTQ5RN2FEU52UUWTXMP5YFA5CNFSM6AAAAAAUW2JW6WWGG33NNVSW45C7OR4XAZNRIRUXGY3VONZWS33OINXW23LFNZ2KUY3PNVWWK3TUL5UWJTQAL5OG4.gif>Message
ID: ***@***.***>

-- 
---8<--
Piotr Karocki

**Reply by YuyhoMarduk** [reply]
basically my app is a brain-computer interface that uses an electroencephalography (eeg) bluetooth device (almost like a headset). I've been searching for the way to do it but it seems like i need some kind of specific uuid for this device.

I went to politechnika krakowska, actually went there few months ago!  Dziękuję!!

**Comment by pkar70** [comment]
If you dont know uuids, you can try ISO standard 11073-something (I cannot
check, Im on vacation). Or, to do some reverse engineering, try my Nuget
pkar.BTdump for UWP
https://www.nuget.org/packages/BTdump

Still, I dont know if you want BT support in Uno, or use plain Android BT
support in your Uno app.


On Thursday, June 22, 2023, YuyhoMarduk ***@***.***> wrote:
> basically my app is a brain-computer interface that uses an
electroencephalography (eeg) bluetooth device (almost like a headset). I've
been searching for the way to do it but it seems like i need some kind of
specific uuid for this device.
>
> I went to politechnika krakowska, actually went there few months ago!
Dziękuję!!
>
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you commented.<
https://ci4.googleusercontent.com/proxy/c9W5Prcf-KTQGXY66pTZsu-ZR1dS2NW67p57IrvQSBhBAmBYUH6TfNXLvoXNNJv62vhol7G1WG_KlhTh3GVC5uFCfgVYNZi6hTJ4q-3I92oM2ufQL0tJwDXI59OmvbFI1jKGnhE3QikxiYWDp7lbuv0PrSt2tbzsm8Kswt-yW4Zu6aQ4McuVQ_KGP8umDfxuzb818fmGPnAE2ifdqkeWTicN8l3RkB7G1U1l13lOTOLdHx5_E2sy81KCvEze=s0-d-e1-ft#https://github.com/notifications/beacon/AFS5O47C32ZJ2QQUJPHWDGTXMREPJA5CNFSM6AAAAAAUW2JW6WWGG33NNVSW45C7OR4XAZNRIRUXGY3VONZWS33OINXW23LFNZ2KUY3PNVWWK3TUL5UWJTQAL5TY6.gif>Message
ID: ***@***.***>

-- 
---8<--
Piotr Karocki
